plugins/modules/zos_job_query.py

```diff
@@ -208,53 +208,86 @@ def run_module():
         return result
 
     try:
-        validate_arguments(module.params)
-        jobs_raw = query_jobs(module.params)
-        jobs = parsing_jobs(jobs_raw)
+        name, id, owner = validate_arguments(module.params)
+        jobs_raw = query_jobs(name, id, owner)
+        if jobs_raw:
+            jobs = parsing_jobs(jobs_raw)
+        else:
+            jobs = None
+
     except Exception as e:
         module.fail_json(msg=e, **result)
     result["jobs"] = jobs
     module.exit_json(**result)
 
 
+# validate_arguments rturns a tuple, so we don't have to rebuild the job_name string
 def validate_arguments(params):
     job_name_in = params.get("job_name")
+
     job_id = params.get("job_id")
+
     owner = params.get("owner")
     if job_name_in or job_id:
         if job_name_in and job_name_in != "*":
             job_name_pattern = re.compile(r"^[a-zA-Z$#@%][0-9a-zA-Z$#@%]{0,7}$")
             job_name_pattern_with_star = re.compile(
                 r"^[a-zA-Z$#@%][0-9a-zA-Z$#@%]{0,6}\*$"
             )
-            m = job_name_pattern.search(job_name_in)
-            n = job_name_pattern_with_star.search(job_name_in)
-            if m or n:
-                pass
-            else:
-                raise RuntimeError("Failed to validate the job name: " + job_name_in)
+            test_basic = job_name_pattern.search(job_name_in)
+            test_star = job_name_pattern_with_star.search(job_name_in)
+            # logic twist: o must be non-null value from m or n
+            test_result = test_basic
+            if test_star:
+                test_result = test_star
+
+            job_name_short = "unused"
+            # if neither m nor n were non-null, check if the string needed to be truncated to the first *
+            if not test_result:
+                ix = job_name_in.find("*")
+                if ix >= 0:
+                    job_name_short = job_name_in[0:ix + 1]
+                    test_result = job_name_pattern.search(job_name_short)
+                    if not test_result:
+                        test_result = job_name_pattern_with_star.search(job_name_short)
+
+            # so now, fail if neither m, n, or o=m/n(short) found a match
+            if not test_result:
+                raise RuntimeError("Failed to validate the job name: " + job_name_in + " ix was " + ix + " short was " + job_name_short)
```

Comment:
@richp405 - is error meant for users , seems very specific to the code, eg `ix was` , users will not know what `ix` is.  I think just returning the value they gave us is enough, no need to share with them the logic of short and ix. 

Wondering if it should be "Unable to find the job name: {0}".format(str(job_name_in))

Reply:
That's simpler and better.  Making the change.